src/components/MMSearch/MMSearch.component.tsx

```diff
@@ -0,0 +1,211 @@
+
+import { Input } from '@Components/Input';
+import { List } from '@Components/List';
+import { AutoCompleteWrapper } from '@Components/AutoComplete/AutoComplete.styles';
+
+import { Constants } from '@Constants';
+
+import { MMSearchProps } from './MMSearch';
+
+/**
+ * MMSearch Component
+ *
+ * @example Correct usage
+ * ```ts
+ * <MMSearch
+ *  searchValue=''
+ *  setSearchValue={(val)=> {}}
+ * 	label='label'
+ *  openOptions={true}
+ *  item={ [
+ * 		{label: 'label 1', value: 'Value 1'},
+ * 		{label: 'label 2', value: 'Value 2'},
+ * 		{label: 'label 3', value: 'Value 3'},
+ * 	]}
+ * />
+ * ```
+ *
+ * @example Correct usage for accessing the selected value
+ * ```ts
+ * <MMSearch
+ *  searchValue=''
+ *  setSearchValue={(val)=> {}}
+ * 	label='label'
+ *  optionsLoading={true}
+ *  fullWidth={true}
+ *  item={[
+ * 		{label: 'label 1', value: 'Value 1', icon: 'User'},
+ * 		{label: 'label 2', value: 'Value 2', icon: 'User'},
+ * 		{label: 'label 3', value: 'Value 3', icon: 'User'},
+ * 	]}
+ * 	onSelect={(event, options)=> {
+ * 		logic to use selected value;
+ * 	}}
+ * />
+ * ```
+ */
+export const MMSearch = (props: MMSearchProps) => {
+    const {
+        fullWidth,
+        items,
+        onSelect,
+        label,
+        className = '',
+        leadingIcon,
+        searchValue,
+        setSearchValue,
+        optionsLoading = false,
+        filterBy = '',
+        openOptions = false,
+        secondaryLabelPosition = null,
+        onClearInput,
+        ...restProps
+    } = props;
+
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const [open, setOpen] = useState<boolean>(openOptions);
+    const [active, setActive] = useState<number>(0);
+
+    const inputRef = useRef<HTMLInputElement>(null);
+    const listRef = useRef<HTMLUListElement>(
+        null,
+    ) as MutableRefObject<HTMLUListElement>;
+
+    /**
+     * On clicking anywhere other than `input field`, the dropdown closes
+     */
+    const onDropDownCloseHandler = (e: MouseEvent) => {
+        e.stopPropagation();
+        if (e.target instanceof HTMLElement && !inputRef.current?.contains(e.target) && e.target !== inputRef.current) {
+            setOpen(false);
+        }
+    };
+
+    // Filter dropdown options based on the input value
+    const filteredOptions = useMemo(
+        () => (filterBy ? items.filter((item) => item.label?.startsWith(filterBy) && item.label !== filterBy) : items),
+        [filterBy, items]);
+
+    useEffect(() => {
+        document.body.addEventListener('click', onDropDownCloseHandler);
+
+        return () => {
+            document.body.removeEventListener('click', onDropDownCloseHandler);
+        };
+    }, []);
+
+    /**
+     * If 'isOpen' is true and 'value' is empty, then set the active index to 0 and scroll the list to (0,0)
+     */
+    useEffect(() => {
+        if (open) {
+            setActive(0);
+            if (typeof listRef.current?.scrollTo === 'function') {
```

Comment:
What is the need to check the type here? Check it everywhere.

Reply:
Removed